daiquiri/oai/adapter.py

```diff
@@ -48,3 +68,149 @@ def get_sample_identifier(self):
+
+    def get_datacite_table_serializer_class(self):
+        return import_class('daiquiri.metadata.serializers.datacite.DataciteTableSerializer')
+
+    def get_schema_list(self):
+        from daiquiri.metadata.models import Schema
+
+        for schema in Schema.objects.iterator():
+            yield 'schema', schema
+
+    def get_table_list(self):
+        from daiquiri.metadata.models import Table
+
+        for table in Table.objects.iterator():
+            yield 'table', table
+
+    def get_schema(self, pk):
+        from daiquiri.metadata.models import Schema
+
+        try:
+            return Schema.objects.get(pk=pk)
+        except Schema.DoesNotExist:
+            return None
+
+    def get_table(self, pk):
+        from daiquiri.metadata.models import Table
+
+        try:
+            return Table.objects.get(pk=pk)
+        except Table.DoesNotExist:
+            return None
+
+    def get_schema_record(self, schema):
+        identifier = self.get_identifier('schemas/%i' % schema.pk)
+        datestamp = schema.updated or schema.published
+        set_spec = 'schema'
+        public = (schema.metadata_access_level == ACCESS_LEVEL_PUBLIC) \
+            and (schema.published is not None)
+
+        return schema.pk, identifier, datestamp, set_spec, public
+
+    def get_table_record(self, table):
+        identifier = self.get_identifier('tables/%i' % table.pk)
+        datestamp = table.updated or table.published
+        set_spec = 'table'
+        public = (table.metadata_access_level == ACCESS_LEVEL_PUBLIC) \
+            and (table.published is not None) \
+            and (table.schema.metadata_access_level == ACCESS_LEVEL_PUBLIC) \
+            and (table.schema.published is not None)
+
+        return table.pk, identifier, datestamp, set_spec, public
+
+
+class DoiMetadataOaiAdapterMixin(MetadataOaiAdapterMixin):
+
+    def get_schema_record(self, schema):
+        identifier = self.get_identifier(schema.doi)
+        datestamp = schema.updated or schema.published
+        set_spec = 'schema'
+        public = (schema.metadata_access_level == ACCESS_LEVEL_PUBLIC) \
+            and bool(schema.published) \
+            and bool(schema.doi)
+
+        return schema.pk, identifier, datestamp, set_spec, public
+
+    def get_table_record(self, table):
+        identifier = self.get_identifier(table.doi)
+        datestamp = table.updated or table.published
+        set_spec = 'table'
+        public = (table.metadata_access_level == ACCESS_LEVEL_PUBLIC) \
+            and bool(table.published) \
+            and bool(table.doi) \
+            and (table.schema.metadata_access_level == ACCESS_LEVEL_PUBLIC) \
+            and bool(table.schema.published)
+
+        return table.pk, identifier, datestamp, set_spec, public
+
+
+class RegistryOaiAdapterMixin(object):
+
+    services = ['registry', 'authority', 'web', 'tap', 'conesearch']
```

Comment:
could we remove `coneserach` and `tap` from there. We have some services that do not have `conesearch` (and usually we activate it at last) and/ or tap (i.e.: xmmssc, cars) and I would prefer to create an adapter where I just need to declare the resources because it is missing, rather than create an adapter because I have too much resources declared.

Reply:
actually, this was a bug. I refactored the file a bit, and now services does not need to be touched. It will check what daiquiri apps are active.